File: backend/app/schemas/analytics.py
from pydantic import BaseModel

from app.schemas.expense import ExpenseResponse
from app.schemas.maintenance import MaintenanceResponse


class ExpenseSummary(BaseModel):
    """High-level expense totals for dashboard cards."""

    total_amount: float
    total_count: int


class MaintenanceSummary(BaseModel):
    """High-level maintenance totals for dashboard cards."""

    total_cost: float
    total_count: int


class CategorySpendItem(BaseModel):
    """Expense spending grouped by category for charts."""

    category_name: str
    total_amount: float
    count: int


class MaintenanceCostByAssetTypeItem(BaseModel):
    """Maintenance cost grouped by asset type for charts."""

    asset_type: str
    total_cost: float
    count: int


class DashboardAnalyticsResponse(BaseModel):
    """Combined dashboard analytics response for the frontend."""

    expenses: ExpenseSummary
    maintenance: MaintenanceSummary
    expense_spend_by_category: list[CategorySpendItem]
    maintenance_cost_by_asset_type: list[MaintenanceCostByAssetTypeItem]
    recent_expenses: list[ExpenseResponse]
    recent_maintenance_records: list[MaintenanceResponse]

File: frontend/src/types/analytics.ts
import type { ExpenseResponse, MaintenanceResponse } from "./domain";

export type { ExpenseResponse, MaintenanceResponse } from "./domain";

export interface ExpenseSummary {
  total_amount: number;
  total_count: number;
}

export interface MaintenanceSummary {
  total_cost: number;
  total_count: number;
}

export interface CategorySpendItem {
  category_name: string;
  total_amount: number;
  count: number;
}

export interface MaintenanceCostByAssetTypeItem {
  asset_type: string;
  total_cost: number;
  count: number;
}

export interface DashboardAnalyticsResponse {
  expenses: ExpenseSummary;
  maintenance: MaintenanceSummary;
  expense_spend_by_category: CategorySpendItem[];
  maintenance_cost_by_asset_type: MaintenanceCostByAssetTypeItem[];
  recent_expenses: ExpenseResponse[];
  recent_maintenance_records: MaintenanceResponse[];
}